cmd/kube-controller-manager/app/options/options_test.go

```diff
@@ -657,6 +656,96 @@ func TestApplyTo(t *testing.T) {
 	}
 }
 
+func TestEmulatedVersion(t *testing.T) {
+	var cleanupAndSetupFunc = func() featuregate.FeatureGate {
+		componentGlobalsRegistry := utilversion.DefaultComponentGlobalsRegistry
+		componentGlobalsRegistry.Reset() // make sure this test have a clean state
+		t.Cleanup(func() {
+			componentGlobalsRegistry.Reset() // make sure this test doesn't leak a dirty state
+		})
```

Comment:
@siyuanfoundation Is this what we will required for tests that modify emulation version, or is only the cleanup reset strictly required?

Reply:
Only the cleanup reset strictly required. But it is also good practice to start with a clean state as well.